apps/reaper/test/integration/reaper/reaper_full_test.exs

```diff
@@ -201,6 +203,30 @@ defmodule Reaper.FullTest do
       end)
     end
 
+    test "configures and ingests a json source using topLevelSelector", %{bypass: bypass} do
+      dataset_id = "topLevelSelectorId"
+      topic = "#{@output_topic_prefix}-#{dataset_id}"
+
+      json_dataset =
+        TDG.create_dataset(%{
+          id: dataset_id,
+          technical: %{
+            cadence: "once",
+            sourceUrl: "http://localhost:#{bypass.port}/#{@json_file_name_subpath}",
+            sourceFormat: "json",
+            topLevelSelector: "$.sub.path"
+          }
+        })
+
+      Brook.Event.send(@instance, dataset_update(), :reaper, json_dataset)
+
+      eventually(fn ->
+        results = TestUtils.get_data_messages_from_kafka(topic, @endpoints)
+
+        assert [%{payload: %{"name" => "Fred"}} | [%{payload: %{"name" => "Bob"}} | _]] = results
+      end)
+    end
+
```

Comment:
Should we add any tests to document some of the non-happy paths?  Failure to parse or failure to find a node that matches the top-level selector?